### ISSUE TITLE
SourceGitHub: Error 500

### ISSUE THREAD

**Issue by Francismori7** [mantisbt-plugins/source-integration#115]
Hello!

I get a 500 error while trying to import my repository. I am running 1.3.0 beta 1, knowing it isn't fully supported, but seeing MantisBT.org uses 1.3.0 beta 2.. I am sure this can work.

I have updated the dependencies (ie changing 1.2.0 to 1.3.0) and also changed the RelatedChangesetsColumn.class.php definitions (to include BugData) so I could view the view all bugs page properly.

But now, whatever I do I get my own repo's name (Esthetica) which I added myself and when running Import all, I get Import 2015-02-02 11:10:57 as well, but is completely empty. I had an error 500 after it hung there for nearly 5 minutes. Checking the repositories page, I get these two empty ones.

**Comment by Francismori7** [comment]
Oh and using master-1.3.x gives me the same result by the way, I had used master first, then tried using the proper branch too.

**Comment by Francismori7** [comment]
Everything suddenly appeared as soon as I commited some data to my github repo.. using the webhook, probably this forced a refresh? (I can't imagine how this would work, and an import wouldn't)

**Comment by dregad** [comment]
Troubleshooting this plugin is a bitch. I've had on my todo list for a long time to integrate proper logging to help with such issues, but never found time to get around to it

See https://github.com/mantisbt-plugins/source-integration/issues/92

**Comment by 01-Scripts** [comment]
May I refresh that one.
I come accross a similar problem.
I'm running Mantis with the Plugin on a shared Webhoster that has runtime restrictions for .php files. After 30s execution time my scripts get stopped with an 500 error.
Now I tried to import my repositories but a full import takes way to long to complete.
Is there any chance to change the import script that way that it imports the data step by step or can be called several times until every data is imported?

**Comment by Francismori7** [comment]
I agree this ended up being my issue as well. We should be able to use ajax
requests on a page to call multiple imports time at a time

**Comment by 01-Scripts** [comment]
Some more information:
Even my configured Webservice doesn't seem to work as long as not all missing changesset where process. At least a timeout warning gets display on the Github site for the mantisbt service.
And if I look in mantis I can see that the changes were transferred but not linked to the matching bugs. The entries in bug_table and changeset_table are missing.

**Comment by 01-Scripts** [comment]
Any chance that we can get a solution for that some time?

**Comment by sverhagen** [comment]
At this point I'm wondering whether the what seems to be timeout can be influenced by PHP settings or Apache settings. It's interesting that @01-Scripts mentions so specifically 30 seconds. Mine is consistently 120 seconds. That's why I think it's some timeout (such as PHP's `max_execution_time`) although I haven't found the right knob to turn.

**Comment by zjhxmjl** [comment]
Hi,guys!had the problem been resolved?i got the same problem,someone can give me some advance?thanks million!

**Comment by zjhxmjl** [comment]
this is error info:
404 Not Found
control "500" not found!
